frontend/src/Pages/FacilityManagement.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+    defaultToast,
+    Facility,
+    ModalType,
+    ServerResponseMany,
+    showToast,
+    ToastState
+} from '@/common.ts';
+import { PlusCircleIcon } from '@heroicons/react/24/outline';
+import { useRef, useState } from 'react';
+import useSWR from 'swr';
+import Toast from '@/Components/Toast.tsx';
+import FacilityCard from '@/Components/FacilityCard.tsx';
+import DeleteForm from '../Components/DeleteForm';
+import Modal from '@/Components/Modal.tsx';
+import AddFacilityForm from '@/Components/forms/AddFacilityForm.tsx';
+import EditFacilityForm from '@/Components/forms/EditFacilityForm';
+import API from '@/api/api';
+import Pagination from '@/Components/Pagination.tsx';
+import { AxiosError } from 'axios';
+
+export default function FacilityManagement() {
+    const addFacilityModal = useRef<HTMLDialogElement>(null);
+    const editFacilityModal = useRef<HTMLDialogElement>(null);
+    const deleteFacilityModal = useRef<HTMLDialogElement>(null);
+    const [editFacility, setEditFacility] = useState<Facility | undefined>();
+    const [toast, setToast] = useState(defaultToast);
+    const [displayToast, setDisplayToast] = useState(false);
+    const toaster = (message: string, toastState: ToastState) => {
+        showToast(setToast, setDisplayToast, message, toastState);
+    };
+    const [targetFacility, setTargetFacility] = useState<
+        undefined | Facility
+    >();
+    // TODO: modify this const
+    const {
+        data: facility,
+        mutate,
+        error,
+        isLoading
+    } = useSWR<ServerResponseMany<Facility>, AxiosError>(`/api/facilities`);
+
+    const facilityData = facility?.data ? facility.data : [];
+    const [, setPageQuery] = useState(1);
```

Comment:
You will have to use this to modify which page in pagination you are on. See the video:

https://github.com/user-attachments/assets/e8b9b730-786a-4920-8b74-a0c79e0385e0